tests/Unit/TestCases/UnitTestCaseTest.php

```diff
@@ -0,0 +1,106 @@
+<?php
+declare(strict_types=1);
+
+namespace Tests\Eonx\TestUtils\Unit\TestCases;
+
+use DateTime;
+use PHPUnit\Framework\AssertionFailedError;
+use PHPUnit\Framework\TestCase;
+use Tests\Eonx\TestUtils\Stubs\TestCases\UnitTestCaseStub;
+
+/**
+ * @covers \Eonx\TestUtils\TestCases\UnitTestCase
+ */
+class UnitTestCaseTest extends TestCase
+{
+    /**
+     * Test array same with dates works fine with date instances.
+     *
+     * @return void
+     *
+     * @throws \Exception
+     */
+    public function testArraySameWithDates(): void
+    {
+        $test = new UnitTestCaseStub();
+
+        $expected = [
+            'random' => 'value',
+            'date' => new DateTime('2019-10-10T00:00:00Z'),
+            'deep' => [
+                'deeper' => [
+                    'date' => new DateTime('2019-10-10T00:00:00+0000'),
+                ]
+            ]
+        ];
+
+        $actual = [
+            'random' => 'value',
+            'date' => new DateTime('2019-10-10T00:00:00Z'),
+            'deep' => [
+                'deeper' => [
+                    'date' => new DateTime('2019-10-10T00:00:00+0000'),
+                ]
+            ]
+        ];
+
+        $test->assertArraySameWithDates($expected, $actual);
```

Comment:
Wouldn't these assertions being tested run inside the test, not be called from outside?

Reply:
oh right, yes `addToAssertionCount` not required here